Incorrect `PathMatcher` Pattern leads to Stack Overflow

### Component

JKube Kit

### Task description

### Task Description

Propose a solution for the problem given below and add a correct exclusion condition for the destination subdirectory (`target/myapp`) for successful copy operation.

### Description

Related To #2733 

While working on the above mentioned issue I have found that an ill designed `PathMatcher` statement is leading to `StackOverflow` in `AssemblyFileSetUtils.copy()`.

![image](https://github.com/eclipse/jkube/assets/111552820/a826f151-ee74-4226-823b-f70efc29a11f)

This is a recursive function and as I suspect it's expected to copy `DockerFile` and  `docker-build.tar` from the output directory (here, `target`) to the subdirectory of the same output directory (here, `target/myapp/build/maven`) in the context of `build_withDockerfileModeAndAssembly_shouldSucceed()`

The exclusion condition for the destination subdirectory is mentioned, here, in `AssemblyManager.createDockerExcludesList()`

![image](https://github.com/eclipse/jkube/assets/111552820/c76b003a-28e2-4d37-936e-fe344e914c31)

But as the `outputDirectory` definition is creating the a incorrect path due to initialization with `target.getAbsolute()`. It creates a incorrect file path in Linux & Windows. In Linux, it leads to the creation of incorrect destination path and in Windows it leads to test failure.

![image](https://github.com/eclipse/jkube/assets/111552820/b922fdc8-85a9-474e-8643-45b5e85efc52)

Destination File Path in Linux:

![image](https://github.com/eclipse/jkube/assets/111552820/9b8fd065-cc44-48bd-b3ce-797f8d87fc3d)

Error in Windows:

![image](https://github.com/eclipse/jkube/assets/111552820/c7c25a6f-b3ba-4a10-a512-607da6614458)

And when I change this `outputDirectory` to `target.getName()` to create the right directory. It leads to `StackOverflow` caused by the infinite recursion due to the incorrect exclusion statement which is not excluding any files or after changing the relativized path leads to exclusion of all files and directories.

![image](https://github.com/eclipse/jkube/assets/111552820/1b76a6b9-56d3-4fb3-9c15-0b5780ce35e7)

### Expected Behavior

As per my analysis, I believe the that the expected behavior is to copy all the files and directories in output directory (`target`) to the subdirectory (for example, `target/myapp/build/maven`) except for the subdirectory itself.

@rohanKanojia & @manusa,  Could you guys please look into this and let me know if I'm right about it.

### Acceptance Criteria
- [ ] `OpenshiftBuildServiceIntegrationTest.build_withDockerfileModeAndAssembly_shouldSucceed()` should pass
- [ ] `AssemblyFileSetUtils.copy()` should lead to successfully copying of files and directories in `target`

Hi @manusa, Sorry for the messy description, There was a lot to cover. The issue persists with both Windows and Linux. 

In Windows:

```org.eclipse.jkube.kit.config.service.JKubeServiceException: Unable to build the image using the OpenShift build service

	at org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.buildSingleImage(OpenshiftBuildService.java:170)
	at org.eclipse.jkube.kit.config.service.AbstractImageBuildService.processImage(AbstractImageBuildService.java:57)
	at org.eclipse.jkube.kit.config.service.AbstractImageBuildService.build(AbstractImageBuildService.java:36)
	at org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildServiceIntegrationTest.build_withDockerfileModeAndAssembly_shouldSucceed(OpenshiftBuildServiceIntegrationTest.java:254)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.util.regex.PatternSyntaxException: Stack overflow during pattern compilation near index 12
^target(?:(?:\\.*)|(?:))$
            ^
	at java.base/java.util.regex.Pattern.error(Pattern.java:2028)
	at java.base/java.util.regex.Pattern.<init>(Pattern.java:1432)
	at java.base/java.util.regex.Pattern.compile(Pattern.java:1095)
	at java.base/sun.nio.fs.WindowsFileSystem.getPathMatcher(WindowsFileSystem.java:282)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.lambda$excludePathMatchers$5(AssemblyFileSetUtils.java:198)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Collections$2.tryAdvance(Collections.java:4853)
	at java.base/java.util.Collections$2.forEachRemaining(Collections.java:4861)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.excludePathMatchers(AssemblyFileSetUtils.java:199)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.isNotExcluded(AssemblyFileSetUtils.java:190)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:166)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
	at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
```


This is occurring due to two reasons, one them is the relativized path for the `source`, which should return the path in `target/<directory-name>` form but returns `<directory name>`

The source code relativization of path is given below `sourceDirectory.relativize(path).normalize()`

![image](https://github.com/eclipse/jkube/assets/111552820/07e2fe4f-bd93-4b82-b8d3-929eec7bab86)

This should return `target/<directory-name>` because the glob pattern created at `AssemblyManager.createDockerExcludesList()` is `target{/**,}` which would remove any files in `target`, which leads to our 2nd issue.

![image](https://github.com/eclipse/jkube/assets/111552820/702de1ac-3304-4938-8857-26da48d09e40)

The second issue is that, we are not supposed to exclude all files in `target` but just the destination subdirectory from it or any directories specified by the user. but due to the above glob `glob:target{/**,}` all the files are being excluded when we change the relativized path to the correct form. 

When all the files are being excluded nothing happens and it never enters the `copy()` block of the code.

The same issue was encountered with Linux but it doesn't return a `StackOverflow` rather it returns:

```
 File name too long
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:100)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixCopyFile.copyFile(UnixCopyFile.java:246)
        at java.base/sun.nio.fs.UnixCopyFile.copy(UnixCopyFile.java:603)
        at java.base/sun.nio.fs.UnixFileSystemProvider.copy(UnixFileSystemProvider.java:257)
        at java.base/java.nio.file.Files.copy(Files.java:1305)
        at org.eclipse.jkube.kit.common.util.FileUtil.copy(FileUtil.java:205)
        at org.eclipse.jkube.kit.common.util.FileUtil.copy(FileUtil.java:201)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:173)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
        at org.eclipse.jkube.kit.common.archive.AssemblyFileSetUtils.copy(AssemblyFileSetUtils.java:170)
``` 

This issue is occurring due to the same reasons mentioned above but here Unix's System call returns error for `File Name Too Long`.

--- a/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildServiceBuildIntegrationTest.java
+++ b/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildServiceBuildIntegrationTest.java
@@ -51,6 +51,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Properties;
 
+import static org.apache.commons.io.FilenameUtils.separatorsToSystem;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
 import static org.mockito.ArgumentMatchers.argThat;
@@ -150,15 +151,15 @@ class JibBuildServiceBuildIntegrationTest {
         .containsExactlyInAnyOrder(
             "Dockerfile",
             "deployments-layer",
-            "deployments-layer/deployments",
-            "deployments-layer/deployments/to-deployments.txt",
+            separatorsToSystem("deployments-layer/deployments"),
+            separatorsToSystem("deployments-layer/deployments/to-deployments.txt"),
             "other-layer",
-            "other-layer/deployments",
-            "other-layer/deployments/other",
-            "other-layer/deployments/other/to-other.txt",
+            separatorsToSystem("other-layer/deployments"),
+            separatorsToSystem("other-layer/deployments/other"),
+            separatorsToSystem("other-layer/deployments/other/to-other.txt"),
             "jkube-generated-layer-final-artifact",
-            "jkube-generated-layer-final-artifact/deployments",
-            "jkube-generated-layer-final-artifact/deployments/final-artifact.jar",
+            separatorsToSystem("jkube-generated-layer-final-artifact/deployments"),
+            separatorsToSystem("jkube-generated-layer-final-artifact/deployments/final-artifact.jar"),
             "deployments"
         );
     ArchiveAssertions.assertThat(dockerDir.resolve("tmp").resolve("docker-build.tar").toFile())

--- a/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildServiceIntegrationTest.java
+++ b/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildServiceIntegrationTest.java
@@ -136,7 +136,7 @@ class OpenshiftBuildServiceIntegrationTest {
         .thenReturn(new ArchiveService(AssemblyManager.getInstance(), logger));
     when(jKubeServiceHub.getBuildServiceConfig().getBuildDirectory()).thenReturn(target.getAbsolutePath());
     when(jKubeServiceHub.getConfiguration()).thenReturn(JKubeConfiguration.builder()
-        .outputDirectory(target.getAbsolutePath())
+        .outputDirectory(target.getName())
         .project(JavaProject.builder()
             .baseDirectory(baseDirectory)
             .buildDirectory(target)
@@ -238,7 +238,7 @@ class OpenshiftBuildServiceIntegrationTest {
   @Test
   void build_withDockerfileModeAndAssembly_shouldSucceed() throws Exception {
     //Given
-    final File dockerFile = new File(target, "Dockerfile");
+    final File dockerFile = new File(baseDirectory, "Dockerfile");
     FileUtils.write(dockerFile, "FROM busybox\n", StandardCharsets.UTF_8);
     image.setBuild(BuildConfiguration.builder()
         .from(projectName)

